client/main/dataset.js

```diff
@@ -118,11 +118,24 @@ const DataSetModel = Backbone.Model.extend({
 
         let columnPB = new coms.Messages.DataSetSchema.ColumnSchema();
         columnPB.id = id;
-        columnPB.name = values.name;
         columnPB.measureType = DataSetModel.parseMeasureType(values.measureType);
 
         let nameChanged = values.name !== column.name;
+        let newName = values.name;
+        let testName = newName;
+        let collision = 0;
         let oldName = column.name;
+        let columns = this.attributes.columns;
+        for (let i = 0; i < columns.length; i++) {
+            if (columns[i].name === testName) {
+                testName = newName + "(" + (collision + 2)  + ")";
+                collision += 1;
+                i = -1;
+            }
+        }
+        newName = testName;
```

Comment:
i don't think this will catch the situation where a subsequent column collides with an earlier one.

i.e.

| column (2) | column | column 3 |

if the person renames `column 3` to be `column`, it will rename it to `column (2)`, even though this clashes with the first column.

i would write this:

    let names = columns.map(column => column.name)
    let testName = name;
    let i = 2;
    while (names.includes(testName))
        testName = name + ' (' + i++ + ')';
    name = testName;

makes sense?